### DISCUSSION TITLE
What's the difference in static vs non-static?

### DISCUSSION THREAD

**Discussion by cocampbe** [maxpert/marmot#88] [Q&A]
I am looking at using marmot with powerdns to replicate the sqlite database. The README only gets me so far. There is little detail on what the difference is in the pacakge that has "static" in the filename? There is a lot of digging/guess work one has to do to get this setup.

**Comment by cocampbe** [comment]
Well, it looks like I cannot use marmot with powerdns. Likely because of a lock.

**Reply by maxpert** [reply]
Would love to get a little more details here. Is the DB locked forever or what are you referring to when saying lock?

**Comment by cocampbe** [comment]
The powerdns process puts a lock on the sqlite database. So marmot can never get a chance to triggwr anything. If I stop powerdns on both servers, marmot works fine.

**Reply by maxpert** [reply]
I wonder why they have to keep DB locked. Have you tried accessing DB via SQLite CLI? Do you get same issue?

**Reply by Habbie** [reply]
powerdns does not keep the db locked

**Comment by cocampbe** [comment]
Yes. The cli works fine. But marmot does not work. I've moved on to another DNS server. So I won't be able to provide much help.

**Reply by maxpert** [reply]
No worries, reads I believe will work but writes I believe won't happen.

**Reply by helander** [reply]
There does not seem to be an answer to the original question:

Difference between static and "non-static" versions

Anyone who knows?

**Reply by maxpert** [reply]
Static binaries of marmot are compiled with static linked lib-musl, while non-static are compiled with standard dynamic libc dependencies
